packages/x-data-grid/src/components/columnHeaders/GridColumnHeaderItem.tsx

```diff
@@ -224,7 +224,7 @@ function GridColumnHeaderItem(props: GridColumnHeaderItemProps) {
       columnMenuButtonId={columnMenuButtonId!}
       field={colDef.field}
       open={columnMenuOpen}
-      target={iconButtonRef.current}
+      target={iconButtonRef}
```

Comment:
Is this change's purpose to avoid rerender when the target element changes? Isn't this a desired behavior?

Reply:
This is already kinda wrong. The source of the value is a ref, which is set by react without triggering a re-render. This was "working" because we are re-rendering compulsively in some places still. I'll think some more how to solve this, but I had a bug with [Popper](https://github.com/mui/mui-x/pull/12980/files#diff-98c7cf5a177f7428296d4c685a7d4d6c7da8c429ffb990421f6ef50cb52aedf8R97-R98) that I tried to hack around.

Reply:
I've thought some more and I think this is the best solution for now, until https://github.com/mui/material-ui/issues/37154 is fixed. It kinda goes against usual react logic, but it's only used internally in a few places.